src/peft/utils/save_and_load.py

```diff
@@ -438,22 +438,38 @@ def load_peft_weights(model_id: str, device: Optional[str] = None, **hf_hub_down
     if device is None:
         device = infer_device()
 
+    def get_hub_filename(use_safetensors=True):
+        weights_name = SAFETENSORS_WEIGHTS_NAME if use_safetensors else WEIGHTS_NAME
+        return (
+            os.path.join(hf_hub_download_kwargs["subfolder"], weights_name)
+            if hf_hub_download_kwargs.get("subfolder", None) is not None
+            else weights_name
+        )
+
     if os.path.exists(os.path.join(path, SAFETENSORS_WEIGHTS_NAME)):
         filename = os.path.join(path, SAFETENSORS_WEIGHTS_NAME)
         use_safetensors = True
     elif os.path.exists(os.path.join(path, WEIGHTS_NAME)):
         filename = os.path.join(path, WEIGHTS_NAME)
         use_safetensors = False
+    elif os.environ.get("HF_HUB_OFFLINE", 0) in {"1", "ON", "YES", "TRUE"}:  # same check as in hfh
```

Comment:
```suggestion
    elif huggingface_hub.constants.HF_HUB_OFFLINE:
```
Better to let `huggingface_hub` handle the env variable parsing. Suggestion needs an additional import.